packages/protocol/contracts/stability/Exchange.sol

```diff
@@ -368,9 +394,38 @@ contract Exchange is
     uint256 updatedStableBucket = exchangeRateNumerator.mul(updatedGoldBucket).div(
       exchangeRateDenominator
     );
+
+    uint256 maxStableBUcketSize = getStableBucketTokenCap();
+
+    // check if the token is bigger than the cap
+    uint256 cappedUpdatedStableBucket = Math.min(maxStableBUcketSize, updatedStableBucket);
+
+    if (cappedUpdatedStableBucket < updatedStableBucket) {
+      // resize gold bucket
+      uint256 cappedUpdatedGoldBucket = exchangeRateDenominator.mul((maxStableBUcketSize)).div(
+        exchangeRateNumerator
+      );
+      return (cappedUpdatedGoldBucket, cappedUpdatedStableBucket);
+    }
+
     return (updatedGoldBucket, updatedStableBucket);
   }
 
+  /**
+   * @notice returns the max size the stableToken can be set durin a bucket update. if the supply of stableToken
+   * is smaller than minSupplyForStableBucketCap, the cap is calculated based on minSupplyForStableBucketCap.
+   */
+  function getStableBucketTokenCap() public view returns (uint256) {
```

Comment:
Again not an expert in naming, but this seems a bit off for me. Wouldn't `getStableTokenBucketCap` be more accurate?